src/overlays/actors/ovl_En_Torch2/z_en_torch2.c

```diff
@@ -27,21 +66,715 @@ const ActorInit En_Torch2_InitVars = {
+                 */
+                if ((player->swordState || (player->actor.velocity.y > -3.0f)) &&
+                    (player->swordAnimation == JUMPSLASH_START)) {                        
+                    this->actor.posRot.rot.y = this->actor.shape.rot.y = this->actor.yawTowardsLink;
+                    
+                    if(globalCtx->gameplayFrames % 2) {
+                        sStickAngle = this->actor.yawTowardsLink + 0x4000;
+                    } else {
+                        sStickAngle = this->actor.yawTowardsLink - 0x4000;
+                    }
+                    sStickTilt = 127.0f; // Does not store with pointer
+                    sJumpslashFlag = false; // Does not store with POinter
+                    input->cur.button |= BTN_A;
+                    sJumpslashTimer = 15;
+                    
+                /*
+                 * Handles jumping on Link's sword
+                 */
+                } else if (sSwordJumpState != 0) {
+                    sStickTilt = 0.0f;
+                    player->stateFlags3 |= 4;
+                    Math_SmoothScaleMaxMinF(&this->actor.posRot.pos.x,
+                                            (Math_Sins(player->actor.shape.rot.y - 0x3E8) * 45.0f) +
+                                                player->actor.posRot.pos.x,
+                                            1.0f, 5.0f, 0.0f);
+                    Math_SmoothScaleMaxMinF(&this->actor.posRot.pos.z,
+                                            (Math_Coss(player->actor.shape.rot.y - 0x3E8) * 45.0f) +
+                                                player->actor.posRot.pos.z,
+                                            1.0f, 5.0f, 0.0f);
+                    sSwordJumpTimer--;
+                    if ((sSwordJumpTimer == 0) || ((player->invincibilityTimer > 0) && !this->swordState)) {
+                        this->actor.posRot.rot.y = this->actor.shape.rot.y = this->actor.yawTowardsLink;
+                        input->cur.button = BTN_A;
+                        player->stateFlags3 &= ~4;
+                        sStickTilt = 127.0f;
+                        player->skelAnime.animCurrentFrame = 3.0f;
+                        sStickAngle = this->actor.yawTowardsLink + 0x8000;
+                        sSwordJumpTimer = sSwordJumpState = 0;
+                        this->actor.flags |= 1;
+                    } else if (sSwordJumpState == 1) {
+                        if (sSwordJumpTimer < 16) {
+                            EnTorch2_SwingSword(globalCtx, input, this);
+                            sSwordJumpState++;
+                        } else if (sSwordJumpTimer == 19) {
+                            func_800F4190(&this->actor.projectedPos, NA_SE_VO_LI_AUTO_JUMP);
+                        }
+                    }
+                } else {
+                    /*
+                     * This does nothing, as sHoldShieldTimer is never set.
+                     */
```

Comment:
this can be a // comment to take up less space

imo all of these can be single line comments, but i understand wanting to make the big ones stand out to break up this very long function. i leave that up to you

Reply:
Yeah, that was definitely intentional. The if tree is very hard to follow without them.

Reply:
i still think a // comment will be just as helpful and be a bit less clutter-ey but its not the biggest deal. can see what others think too

Reply:
I agree with Fig here. In general, I prefer leaving multi line comments to function/file level documentation, so it's easier to distinguish them when going through the file. 

Also, as a small related note, the style for multi line comments is to start with `/**` on the first line, and always break before and after the comment's body. VSCode (like most IDEs) will even "understand" it's a multi line comment if you start it with `/**` then press enter to start typing.

Reply:
Feel free to leave more spaces between blocks if you need to tho. I don't mention it in reviews, but nowadays I think a lot of clarity could be gained in the codebase by leaving empty lines in the right places. For example here I would leave an empty line after each switch case and after each if block, at the very least.